File: src/main/java/cabalgatas_salento/reservas/mapper/SalidaMapper.java
package cabalgatas_salento.reservas.mapper;

import cabalgatas_salento.reservas.dto.response.SalidaResponse;
import cabalgatas_salento.reservas.entity.Salida;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Component;

@Component
@RequiredArgsConstructor
public class SalidaMapper {

    private final RutaMapper rutaMapper;
    private final CaballoMapper caballoMapper;
    private final GuiaMapper guiaMapper;

    public SalidaResponse toResponse(Salida salida) {
        return SalidaResponse.builder()
                .id(salida.getId())
                .ruta(rutaMapper.toResponse(salida.getRuta()))
                .fechaProgramada(salida.getFechaProgramada())
                .tiempoInicio(salida.getTiempoInicio())
                .tiempoFin(salida.getTiempoFin())
                .estado(salida.getEstado())
                .caballos(salida.getCaballos().stream()
                        .map(sc -> caballoMapper.toResponse(sc.getCaballo()))
                        .toList())
                .guias(salida.getGuias().stream()
                        .map(sg -> guiaMapper.toResponse(sg.getGuia()))
                        .toList())
                .build();
    }
}

File: src/salidas/mappers/salida.mapper.ts
import { CaballoMapper } from '../../caballos/mappers/caballo.mapper';
import { GuiaMapper } from '../../guias/mappers/guia.mapper';
import { RutaMapper } from '../../rutas/mappers/ruta.mapper';
import { SalidaResponseDto } from '../dto/salida-response.dto';
import { Salida } from '../entities/salida.entity';

export class SalidaMapper {
  static toResponseDto(entity: Salida): SalidaResponseDto {
    const dto = new SalidaResponseDto();
    dto.id = entity.id;
    dto.ruta = entity.ruta ? RutaMapper.toResponseDto(entity.ruta) : null;
    dto.fechaProgramada = entity.fechaProgramada;
    dto.tiempoInicio = entity.tiempoInicio;
    dto.tiempoFin = entity.tiempoFin;
    dto.estado = entity.estado;
    dto.caballos = (entity.caballos ?? []).map((sc) =>
      CaballoMapper.toResponseDto(sc.caballo),
    );
    dto.guias = (entity.guias ?? []).map((sg) =>
      GuiaMapper.toResponseDto(sg.guia),
    );
    return dto;
  }
}
